src/engines/html5/html5.js

```diff
@@ -208,19 +208,31 @@ export default class Html5 extends FakeEventTarget implements IEngine {
    * @returns {void}
    */
   reset(): void {
+    if (this._reset) return;
+    this._reset = true;
     this._eventManager.removeAll();
     if (this._droppedFramesWatcher) {
       this._droppedFramesWatcher.destroy();
       this._droppedFramesWatcher = null;
     }
-    if (this._mediaSourceAdapter) {
-      this._canLoadMediaSourceAdapterPromise = this._mediaSourceAdapter.destroy();
-      this._mediaSourceAdapter = null;
-    }
-    if (this._el && this._el.src) {
-      Utils.Dom.setAttribute(this._el, 'src', '');
-      Utils.Dom.removeAttribute(this._el, 'src');
-    }
+    this._canLoadMediaSourceAdapterPromise = new Promise((resolve, reject) => {
+      const mediaSourceAdapterDestroyed = this._mediaSourceAdapter ? this._mediaSourceAdapter.destroy() : Promise.resolve();
+      if (this._el && this._el.src) {
+        mediaSourceAdapterDestroyed.then(
+          () => {
+            Utils.Dom.setAttribute(this._el, 'src', '');
+            Utils.Dom.removeAttribute(this._el, 'src');
+            resolve();
+          },
+          () => {
+            reject();
+          }
+        );
+      } else {
+        resolve();
+      }
+    });
+    this._mediaSourceAdapter = null;
```

Comment:
we are still not waiting for the `mediaSourceAdapterDestroyed` to be resolved then in case there is no `src`.
Not sure if it matters or not but it is not consistent in terms of flow control.

Reply:
why not? if I don't have src it could be handled on adapters or even 3rd party libs, bottom line I'm in a state of no src on video tag same as destroy, so for the app it's the same spot as destroyed completely.

Reply:
More around consistency than anything else, can you explain why inconsistency here is better? does it serve a purpose?

Reply:
it keeps the previous logic, before if src didn't exist it didn't stop the load or anything else, just did nothing...

Reply:
Here if it'll reject we'll stop the load process.

Reply:
the old logic obviously wasn't the poet's intention - the promise was always meant to be the shaka destroy promise from playkit-js-dash

Reply:
right, so I kept this logic...
I don't think that if we have or haven't source should influence the destroy state.

Reply:
It has nothing to do with src.
issue is that the the `_canLoadMediaSourceAdapterPromise` should only be resolved once the media adapter promise ends.
Now we are still short circuiting it if we immediately resolve it and not wait for the adapter to resolve!